### PR TITLE
Re-run last query with bare `\watch`

### LINKED ISSUE #544: \watch command does not work
(this is a minor issue as `\watch` is not a well known command)

PostgreSQL has a `\watch` command to repeat queries every X seconds (as documented here): https://www.postgresql.org/docs/current/static/app-psql.html

It works perfectly on the psql CLI:

``` sql
gab@db > SELECT 1; \watch 1;
 ?column? 
----------
        1
(1 row)

Time: 0.187 ms
Watch every 1s  Wed Jul  6 12:01:54 2016

 ?column? 
----------
        1
(1 row)

Time: 0.047 ms
Watch every 1s  Wed Jul  6 12:01:55 2016

 ?column? 
----------
        1
(1 row)

Time: 0.154 ms
Watch every 1s  Wed Jul  6 12:01:56 2016

 ?column? 
----------
        1
(1 row)

Time: 0.164 ms
```

It does not work with pgcli:

``` sql
db> SELECT 1; \watch 1;
+------------+
|   ?column? |
|------------|
|          1 |
+------------+
SELECT 1
syntax error at or near "\"
LINE 1: \watch 1
        ^

Time: 0.003s
```

### PR DESCRIPTION
**Pull request by saifelse**
## Description
Like `psql`, allow a user to first execute a query, and then run `\watch` to
re-run the query in a loop.

The remaining parity folks were looking for in #544 will be handled by
dbcli/pgspecial#120 , which is a non-breaking API change and should
thus already be compatible with the latest version of `pgcli`.

Closes #544

### Example:
```sql
$ pgcli -h localhost -p 5432 -U pg db
Server: PostgreSQL 11.9 (Debian 11.9-1.pgdg90+1)
Version: 3.2.0
Home: http://pgcli.com
pg@localhost:db> \watch 1
\watch cannot be used with an empty query

pg@localhost:db> select 123;
+------------+
| ?column?   |
|------------|
| 123        |
+------------+
SELECT 1
Time: 0.018s
pg@localhost:db> \watch
+------------+
| ?column?   |
|------------|
| 123        |
+------------+
SELECT 1
Time: 0.012s
Waiting for 2 seconds before repeating
+------------+
| ?column?   |
|------------|
| 123        |
+------------+
SELECT 1
Time: 0.015s
Waiting for 2 seconds before repeating
^C pg@localhost:db> \watch 1
+------------+
| ?column?   |
|------------|
| 123        |
+------------+
SELECT 1
Time: 0.013s
Waiting for 1 seconds before repeating
+------------+
| ?column?   |
|------------|
| 123        |
+------------+
SELECT 1
Time: 0.018s
Waiting for 1 seconds before repeating
^C pg@localhost:db>
```

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

- [x] I installed pre-commit hooks (`pip install pre-commit && pre-commit install`), and ran `black` on my code.
- [x] Please squash merge this pull request (uncheck if you'd like us to merge as multiple commits)

### CODE REVIEW

File: pgcli/main.py [hunk 1]
```diff
@@ -785,6 +774,36 @@ def run_cli(self):
...
+
+        # If we run \watch without a command, apply it to the last query run.
+        if self.watch_command is not None and not self.watch_command.strip():
+            try:
+                self.watch_command = self.query_history[-1].query
+            except IndexError:
+                click.secho(
+                    "\\watch cannot be used with an empty query", err=True, fg="red"
+                )
+                self.watch_command = None
```
Comment (saifelse) [inline pgcli/main.py:789]: this is the new logic

**Review by j-bennet — APPROVED** [review]
Nice work! Please resolve the conflicts, so I can merge this.

**Comment by saifelse** [comment]
@j-bennet : done!

**Comment by j-bennet** [comment]
@saifelse I also sent you an email a few days ago, have you happened to read it yet? it was from i.chernyavska_at_gmail_com.

**Pull request merged**